### PR TITLE
Support sorting by time desc

### LINKED ISSUE #2022: Wire up ORDER BY
```
> select statusCode from SeriesName order by time desc limit 10
name: HttpRequestRate
---------------------
time				statusCode
2015-03-19T20:09:14.667313352Z	200
2015-03-19T20:09:14.667531825Z	200
2015-03-19T20:09:14.674292103Z	200
2015-03-19T20:09:14.674539615Z	200
2015-03-19T20:09:23.334009887Z	200
2015-03-19T20:09:23.334241444Z	200
2015-03-19T20:09:23.33466066Z	200
2015-03-19T20:09:38.333210425Z	200
2015-03-19T20:09:38.333536172Z	200
2015-03-19T20:09:38.334099221Z	200

> select statusCode from SeriesName order by time asc limit 10
name: HttpRequestRate
---------------------
time				statusCode
2015-03-19T20:09:14.667313352Z	200
2015-03-19T20:09:14.667531825Z	200
2015-03-19T20:09:14.674292103Z	200
2015-03-19T20:09:14.674539615Z	200
2015-03-19T20:09:23.334009887Z	200
2015-03-19T20:09:23.334241444Z	200
2015-03-19T20:09:23.33466066Z	200
2015-03-19T20:09:38.333210425Z	200
2015-03-19T20:09:38.333536172Z	200
2015-03-19T20:09:38.334099221Z	200
```

### PR DESCRIPTION
**Pull request by jwilder**
This PR adds support for sorting results by time in descending order.  Two syntaxes are allowed:

```
select value from cpu order by time desc
select value from cpu order by desc
```

This allows queries to find the most recent value instead of having to query the whole data set to see the last value.  Currently only sorting by time is allowed on raw and aggregate queries.  If there are multiple sort fields, a parse error will be returned.  Similarly, if a sort field other than `time` is specified, a parse error is returned.

Fixes #2022

### CODE REVIEW

**Comment by otoolep** [comment]
Cool.

Are all the `true` and `false` required in the tests? Are there not enums for those?

**Comment by jwilder** [comment]
Looks like I forgot to switch those over to the enum.  Will fix.

**Comment by otoolep** [comment]
Should we expect a query performance hit for normal time-ascending queries? You should probably check it, if you haven't already.

File: cmd/influxd/run/server_test.go [hunk 1]
```diff
@@ -2201,6 +2201,14 @@ func TestServer_Query_Aggregates(t *testing.T) {
 			command: `SELECT sum(value)/2 FROM load`,
 			exp:     `{"results":[{"series":[{"name":"load","columns":["time",""],"values":[["1970-01-01T00:00:00Z",75]]}]}]}`,
 		},
+
+		// order by time desc
+		&Query{
+			name:    "order by time desc",
```
Comment (corylanou) [inline cmd/influxd/run/server_test.go]: maybe `aggregate order by time desc`?

**Comment by corylanou** [comment]
Very nice.  Great test coverage.  +1

**Comment by jwilder** [comment]
@otoolep Performance for raw and aggregate queries for asc and desc is the same from master to my branch using 10M points.  Desc doesn't seem to have much overhead either which is good.

**Pull request merged**